File: AoC-2024/Day-02/2024-02-1.py
with open('Input.txt', 'r') as f:
    txt = f.read()
txt_lines = txt.split('\n')

ans = 0
for line in txt_lines:
    line = [int(i) for i in line.split()]

    #Line has only 1 report
    if len(line) == 1:
        ans += 1
    
    diffs = [line[i+1]-line[i] for i in range(len(line)-1)]
    m,M = min(diffs),max(diffs)
    if 1 <= m and M <= 3:
        ans += 1
    if -3 <= m and M <= -1:
        ans += 1

print(ans)

File: AoC-2024/Day-02/2024-02-1.js
const fs = require('fs')
txt = fs.readFileSync('Input.txt', 'utf8')

lines = txt.split('\n')

ans = 0
for (Homu = 0; Homu < lines.length; Homu++) {
    line = lines[Homu]
    a = line.split(' ')
    diffs = []
    for (i = 0; i < a.length-1; i++) {
        diffs[i] = a[i+1]-a[i]
    }
    m = Math.min(...diffs)
    M = Math.max(...diffs)
    if (-3 <= m && M <= -1) {
        ans += 1
    }
    if (1 <= m && M <= 3) {
        ans += 1
    }
}
console.log(ans)
